Cleanup dependency check; fixes #977

trasher opened the pull request.

remicollet commented on inc/toolbox.class.php:
```diff
@@ -941,111 +927,87 @@ static function commonCheckForUseGLPI() {
+         'mysqli'   => [
+            'title'     => __('MySQL Improved extension test'),
+            'success'   => __s('Ok - the MySQLi class exist - Perfect!'),
+            'error'     => __('You must install the MySQL Improved extension for PHP.')
```
I think this should be 1 or 2 (the error return)
Then same message can be used for all

remicollet replied:
I mean 1=optional, 2=mandatory (see install.php)

trasher replied:
Indeed, we got a `$error = 1;` for the `gd` extension test; I did not see it. thank you!

remicollet commented on inc/toolbox.class.php:
```diff
@@ -941,111 +927,87 @@ static function commonCheckForUseGLPI() {
+                    title=\"{$params['success']}\"></td>";
+         } else {
+            echo "<td class=\"red\"><img src=\"{$CFG_GLPI['root_doc']}/pics/ko_min.png\">{$params['error']}</td>";
+            $error = 2;
```
See previous.

if ($error=1) ....'you should install '
if ($error=2) ....'you MUST install '

remicollet commented on inc/toolbox.class.php:
```diff
@@ -941,111 +927,87 @@ static function commonCheckForUseGLPI() {
+            'success'   => __s('The functionality is found - Perfect!'),
+            'error'     => __('Curl extension of your parser PHP is not installed')
+         ]
+      ];
```
Il est donc possible d'ajouter imap et ldap qui sont optionnelles (comme gd)

remicollet commented on inc/toolbox.class.php:
```diff
@@ -941,111 +927,87 @@ static function commonCheckForUseGLPI() {
+         $success = true;
+
+         if (isset($params['functions'])) {
+            foreach ($params['functions'] as $function) {
```
I'm not sure testing various functions is really useful, a single one should be enough.

trasher commented:
So... Should we merge this one for 9.1; or wait for the next release?

remicollet commented on inc/toolbox.class.php:
```diff
@@ -941,111 +927,77 @@ static function commonCheckForUseGLPI() {
-                     __s('The functionality is found - Perfect!')."\"></td>";
+         echo "<tr class=\"tab_bg_1\"><td class=\"left b\">{sprintf('%s extension test', $ext)}</td>";
+         if ($success === true) {
+            $msg = sprintf('Extension %s is installed', $ext);
```
gettext is missing

trasher replied:
Hey... Well catched, thanks :)

remicollet commented on inc/toolbox.class.php:
```diff
@@ -941,111 +927,77 @@ static function commonCheckForUseGLPI() {
+         } else {
+            if (isset($params['required']) && $params['required'] === true) {
+               $error = 2;
+               echo "<td class=\"red\"><img src=\"{$CFG_GLPI['root_doc']}/pics/ko_min.png\">{sprintf('Extension %s is missing', $ext)}</td>";
```
Also missig here ;)

remicollet commented on inc/toolbox.class.php:
```diff
@@ -941,111 +927,77 @@ static function commonCheckForUseGLPI() {
+               echo "<td class=\"red\"><img src=\"{$CFG_GLPI['root_doc']}/pics/ko_min.png\">{sprintf('Extension %s is missing', $ext)}</td>";
+            } else {
+               $error = 1;
+               echo "<td><img src=\"{$CFG_GLPI['root_doc']}/pics/warning_min.png\">{sprintf('Extension %s is not present', $ext)}</td>";
```
and here :p

remicollet commented on inc/toolbox.class.php:
```diff
@@ -941,111 +927,77 @@ static function commonCheckForUseGLPI() {
-         echo "<td><img src='".$CFG_GLPI['root_doc']."/pics/ok_min.png' alt=\"".
-                     __s('The functionality is found - Perfect!'). "\" title=\"".
-                     __s('The functionality is found - Perfect!')."\"></td>";
+         echo "<tr class=\"tab_bg_1\"><td class=\"left b\">{sprintf(__('%s extension test'), $ext)}</td>";
```
doesn't work, I think you cannot escape function call in this way, only var.

trasher replied:
I've fixed that, thanks :)

trasher commented:
OK, ths is not corect. `$error` can be overriden... Will have to rework.

trasher commented:
OK, I've reworked that... I had inverted a true and false (so, yes, that was not working as expected :D).

I've also changed string to "%s extension" for all.

remicollet commented:
Thanks for your patience ;)

Merged (I just add a space between the icon and the message)

trasher commented:
Thanks for your patience :D
